python/cugraph/cugraph/dask/community/induced_subgraph.py

```diff
@@ -0,0 +1,199 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from dask.distributed import wait, default_client
+
+import cugraph.dask.comms.comms as Comms
+import dask_cudf
+import cudf
+from cugraph.dask.common.input_utils import get_distributed_data
+
+from pylibcugraph import (
+    ResourceHandle,
+    induced_subgraph as pylibcugraph_induced_subgraph,
+)
+
+
+def _call_induced_subgraph(
+    sID,
+    mg_graph_x,
+    vertices,
+    offsets,
+    do_expensive_check,
+):
+    return pylibcugraph_induced_subgraph(
+        resource_handle=ResourceHandle(Comms.get_handle(sID).getHandle()),
+        graph=mg_graph_x,
+        subgraph_vertices=vertices,
+        subgraph_offsets=offsets,
+        do_expensive_check=do_expensive_check,
+    )
+
+
+def consolidate_results(df, offsets):
+    """
+    Each rank returns its induced_subgraph dataframe with its corresponding
+    offsets array. This is ideal if the user operates on distributed memory
+    but when attempting to bring the result into a single machine,
+    the induced_subgraph dataframes generated from each seed cannot be extracted
+    using the offsets array. This function consolidate the final result by
+    performing segmented copies.
+
+    Returns: consolidated induced_subgraph dataframe
+    """
+    for i in range(len(offsets) - 1):
+        df_tmp = df[offsets[i] : offsets[i + 1]]
+        df_tmp["labels"] = i
+        if i == 0:
+            df_consolidate = df_tmp
+        else:
+            df_consolidate = cudf.concat([df_consolidate, df_tmp])
+    return df_consolidate
+
+
+def convert_to_cudf(cp_arrays):
+    cp_src, cp_dst, cp_weight, cp_offsets = cp_arrays
+
+    df = cudf.DataFrame()
+    df["src"] = cp_src
+    df["dst"] = cp_dst
+    df["weight"] = cp_weight
+
+    offsets = cudf.Series(cp_offsets)
+
+    return consolidate_results(df, offsets)
+
+
+def induced_subgraph(input_graph, vertices, offsets=None):
+    """
+    Compute a subgraph of the existing graph including only the specified
+    vertices.  This algorithm works with both directed and undirected graphs
+    and does not actually traverse the edges, but instead simply pulls out any
+    edges that are incident on vertices that are both contained in the vertices
+    list.
+
+    If no subgraph can be extracted from the vertices provided, a 'None' value
+    will be returned.
+
+    Parameters
+    ----------
+    input_graph : cugraph.Graph
+        Graph or matrix object, which should contain the connectivity
+        information. Edge weights, if present, should be single or double
+        precision floating point values.
+
+    vertices : cudf.Series or cudf.DataFrame
+        Specifies the vertices of the induced subgraph. For multi-column
+        vertices, vertices should be provided as a cudf.DataFrame
+
+    offsets : cupy array
```

Comment:
Is the type here supposed to be `cudf.Series` ? Just asking because you create a `cudf.Series` if `offsets` is None below.